mlmodel/docs/Format/Model.rst

```diff
@@ -1,112 +1,226 @@
-	    // If not supplied it defaults to ["model0",..., "model"(models.size()-1)]
-	    // These names can be used to disambiguate the scope / domain of a parameter
-	    repeated string names = 2;
-	}
-
+Data Structures, Feature Types, and Parameters
+  - `DataStructures`
+  - `FeatureTypes`
+  - `Parameters`
 
 
 
+Model
+________________________________________________________________________________
 
 
-PipelineClassifier
-________________________________________________________________________________
+Core ML model compatibility is indicated by
+a monotonically increasing specification version number,
+which is incremented anytime a backward-incompatible change is made
+(this is functionally equivalent to the MAJOR version number
+described by `Semantic Versioning 2.0.0 <http://semver.org/>`_).
 
-A classifier pipeline.
+Specification Versions : OS Availability (Core ML Version)
 
+1 : iOS 11, macOS 10.13, tvOS 11, watchOS 4 (Core ML 1)
 
-.. code-block:: proto
+	- Feedforward & Recurrent Neural Networks
+	- General Linear Models
+	- Tree Ensembles
+	- Support Vector Machines
+	- Pipelines
+	- Feature Engineering
 
-	message PipelineClassifier {
-	    Pipeline pipeline = 1;
-	}
+2 : iOS 11.2, macOS 10.13.2, tvOS 11.2, watchOS 4.2 (Core ML 1.2)
 
+	- Custom Layers for Neural Networks
+	- Float 16 support for Neural Network layers
 
+3 : iOS 12, macOS 10.14, tvOS 12, watchOS 5 (Core ML 2)
 
+	- Flexible shapes and image sizes
+	- Categorical sequences
+	- Core ML Vision Feature Print, Text Classifier, Word Tagger
+	- Non Max Suppression
+	- Crop and Resize Bilinear NN layers
+	- Custom Models
 
+4 : iOS 13, macOS 10.15, tvOS 13, watchOS 6 (Core ML 3)
 
+	- Updatable models
+	- Exact shape / general rank mapping for neural networks
+	- Large expansion of supported neural network layers:
 
-PipelineRegressor
-________________________________________________________________________________
+		 - Generalized operations
+		 - Control flow
+		 - Dynamic layers
+		 - See NeuralNetwork.proto
+	
+	- Nearest Neighbor Classifier
+	- Sound Analysis Prepreocessing
+	- Recommender
+	- Linked Model
+	- NLP Gazeteer
+	- NLP WordEmbedding
+
+5 : iOS 14, macOS 11, tvOS 14, watchOS 7 (Core ML 4)
+
+	- Model Deployment
+	- Model Encryption
+	- Unified converter API with PyTorch and Tensorflow 2 Support in coremltools 4
+	- MIL builder for neural networks and composite ops in coremltools 4
+	- New layers in neural network:
+	
+		 - CumSum
+		 - OneHot
+		 - ClampedReLu
+		 - ArgSort
+		 - SliceBySize
+		 - Convolution3D
+		 - Pool3D
+		 - Bilinear Upsample with align corners and fractional factors
+		 - PixelShuffle
+		 - MatMul with int8 weights and int8 activations
+		 - Concat interleave
+		 - See NeuralNetwork.proto
+	
+	- Enhanced Xcode model view with interactive previews
+	- Enhanced Xcode Playground support for Core ML models
 
-A regressor pipeline.
+6 : iOS 15, macOS 12, tvOS 15, watchOS 8 (Core ML 5)
 
+	- Core ML Audio Feature Print
```

Comment:
Another bullet is missing here: MIL model type

Reply:
OK, made the change, will be added in the next commit.

Reply:
@aseemw , should "MIL" be included at the top of `Model.rst`, as a bullet under "Neural Networks"?